skimage/transform/_geometric.py

```diff
@@ -970,8 +1047,64 @@ def inverse(self, coords):
         return out
 
 
+def _euler_rotation(axis, angle):
+    """Produce a single-axis Euler rotation matrix.
+
+    Parameters
+    ----------
+    axis : int in {0, 1, 2}
+        The axis of rotation.
+    angle : float
+        The angle of rotation in radians.
+
+    Returns
+    -------
+    Ri : array of float, shape (3, 3)
+        The rotation matrix along axis `axis`.
+    """
+    i = axis
+    s, c = np.sin(angle), np.cos(angle)
+    R2 = np.array([[          c, (-1)**(i+1) * s],
+                   [(-1)**i * s,               c]])
+    Ri = np.eye(3)
+    # We need the axes other than the rotation axis, in the right order:
+    # 0 -> (1, 2); 1 -> (0, 2); 2 -> (0, 1).
+    axes = sorted({0, 1, 2} - {axis})
+    # We then embed the 2-axis rotation matrix into the full matrix.
+    # (1, 2) -> R[1:3:1, 1:3:1] = R2, (0, 2) -> R[0:3:2, 0:3:2] = R2, etc.
+    sl = slice(axes[0], axes[1] + 1, axes[1] - axes[0])
+    Ri[sl, sl] = R2
+    return Ri
+
+
+def _euler_rotation_matrix(angles, axes=None):
+    """Produce an Euler rotation matrix from the given angles.
+
+    The matrix will have dimension equal to the number of angles given.
+
+    Parameters
+    ----------
+    angles : array of float, shape (3,)
+        The transformation angles in radians.
+    axes : list of int
+        The axes about which to produce the rotation. Defaults to 0, 1, 2.
+
+    Returns
+    -------
+    R : array of float, shape (3, 3)
+        The Euler rotation matrix.
+    """
+    if axes is None:
+        axes = list(range(3))
```

Comment:
```suggestion
        axes = tuple(range(3))
```
It doesn't look like the list is modified later, so may as well use a tuple

Reply:
See option c :joy: 

https://twitter.com/ashwintsrinath/status/1339001641752801280

Reply:
actually, we only iterate over `axes` later, so I'm just gonna leave it as `range()`!